### PR TITLE
Cashier app improved to show available quantity to sell for a specific date

### PR DESCRIPTION
**Pull request by diliger**
according to transfers that were before the date and sells that were
before and after the date.

### CODE REVIEW

**Comment by diliger** [comment]
This improvement requires updated `Simplified`.
Calculated property `ProductInstance.Transfer` was added.

File: src/Cashier/Helpers/Organizations.cs [hunk 1]
```diff
@@ -37,27 +37,31 @@ public static Organization GetCustomer()
...
+            return Db.SQL<Product>("SELECT P FROM Simplified.Ring3.Product P").Where(val => val.GetStockInShop(Shop, ForDate) > 0);
         }
 
         /// <summary>
         /// Get current stock of given product in a given organization
         /// </summary>
         /// <param name="Shop">Stock will be calculated only for this organization</param>
         /// <param name="Product">Stock will be calculated for this product only</param>
-        public static decimal GetStockInShop(this Organization Shop, Product Product)
+        public static decimal GetStockInShop(this Organization Shop, Product Product, DateTime? ForDate)
```
Comment (joozek78) [inline src/Cashier/Helpers/Organizations.cs]: Please add docstring for new parameter

Comment (joozek78) [inline src/Cashier/Helpers/Organizations.cs]: Why is `ForDate` nullable? If somebody wants the stock for today, let him express by passing Date.Now

Comment (diliger) [inline src/Cashier/Helpers/Organizations.cs]: Fixed both.

Comment (joozek78) [inline src/Cashier/Helpers/Organizations.cs]: I don't see the changes for nullable parameters. This pertains to all methods with DateTime? here

**Comment by diliger** [comment]
Yes, found it, there are multiple methods with that name.
Fixed.

**Pull request merged**